src/test/resources/size-based-aggregators-config.xml

```diff
@@ -91,6 +93,59 @@
         </aggregators:size-based-aggregator>
     </flow>
 
+    <flow name="onIncrementalIdCheck">
+        <aggregators:size-based-aggregator name="8" content="#[payload]" maxSize="3">
+            <aggregators:incremental-aggregation>
+                <choice>
+                    <when expression="#[vars.id != null]">
+                        <test:assert expression="#[vars.id == attributes.aggregationId]"/>
+                    </when>
+                    <otherwise>
+                        <set-variable variableName="id" value="#[attributes.aggregationId]"/>
+                    </otherwise>
+                </choice>
+            </aggregators:incremental-aggregation>
+            <aggregators:aggregation-complete>
+                <test:processor/>
+            </aggregators:aggregation-complete>
+        </aggregators:size-based-aggregator>
+    </flow>
+
+    <flow name="onCompleteAndListenerIdCheck">
+        <aggregators:size-based-aggregator name="9" content="#[payload]" maxSize="2">
+            <aggregators:incremental-aggregation>
+                <set-variable variableName="id" value="#[attributes.aggregationId]"/>
+            </aggregators:incremental-aggregation>
+            <aggregators:aggregation-complete>
+                <test:assert expression="#[vars.id == attributes.aggregationId]"/>
+            </aggregators:aggregation-complete>
+        </aggregators:size-based-aggregator>
+    </flow>
+
+    <flow name="idCheckListener">
+        <aggregators:aggregator-listener aggregatorName="9"/>
+        <set-payload value="#[correlationId]"/>
+        <flow-ref name="onListener"/>
+    </flow>
+
+    <flow name="idChangeAfterComplete">
+        <aggregators:size-based-aggregator name="10" content="#[payload]" maxSize="2">
```

Comment:
content must be a child attribute and in the SDK must be annotated with @Content. Please review all other content parameters.